packages/grid/_modules_/grid/hooks/features/selection/gridSelectionSelector.ts

```diff
@@ -17,9 +17,7 @@ export const selectedGridRowsSelector = createSelector(
     new Map<GridRowId, GridRowModel>(selectedRows.map((id) => [id, rowsLookup[id]])),
 );
 
-export const selectedIdsLookupSelector = createSelector(gridSelectionStateSelector, (selection) =>
-  selection.reduce((lookup, rowId) => {
-    lookup[rowId] = rowId;
-    return lookup;
-  }, {}),
+export const selectedIdsLookupSelector = createSelector(
+  gridSelectionStateSelector,
+  (selection) => new Map<GridRowId, true>(selection.map((el) => [el, true])),
```

Comment:
My main issue with the previous behavior is that the keys of an object are casted to strings which can cause bugs when checking id equality. I had the issue here https://github.com/mui-org/material-ui-x/pull/2561/files#diff-5c6a5cdf37215fa48d353e1449ef8cc1768d4a4a92f28e4f223c9ba325952a4cR139 for instance.

But I can revert for this PR

Reply:
Using an Object has a little performance advantage over the Map: https://jsbench.me/ijkq9pwctd/3

If I understood, the problem you found in https://github.com/mui-org/material-ui-x/pull/2561/files#diff-5c6a5cdf37215fa48d353e1449ef8cc1768d4a4a92f28e4f223c9ba325952a4cR139 is not related to selection, right?

Reply:
If their is a performance advantage I should definitely revert.

No, the problem I found in the link you gave is not on the same object.
But the idea is the same. 
If we keep the "object with the ID as key" pattern, then we must be careful not to extract the Ids from this object because it introduces a type casting to string, which can cause nasty bugs.

Reply:
> If their is a performance advantage I should definitely revert.

👍

> If we keep the "object with the ID as key" pattern, then we must be careful not to extract the Ids from this object because it introduces a type casting to string, which can cause nasty bugs.

Yes, the idea is: use the key to check if it's present, but use the value as the id.